src/Pelagos/Bundle/AppBundle/Controller/UI/SideBySideController.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Pelagos\Bundle\AppBundle\Controller\UI;
+
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Method;
+
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+use Pelagos\Bundle\AppBundle\Form\DatasetSubmissionType;
+
+use Pelagos\Entity\Account;
+use Pelagos\Entity\DatasetSubmission;
+use Pelagos\Entity\Dataset;
+use Pelagos\Entity\DIF;
+use Pelagos\Entity\PersonDatasetSubmissionDatasetContact;
+
+use Pelagos\Response\TerminateResponse;
+
+/**
+ * The DIF controller for the Pelagos UI App Bundle.
+ *
+ * @Route("/sidebyside")
+ */
+class SideBySideController extends UIController
+{
+    /**
+     * Valid values for $datasetFileTransferType and $metadataFileTransferType.
+     */
+    const SUBMISSIONS_STATES = array(
+        DatasetSubmission::STATUS_UNSUBMITTED => 'Unsubmitted',
+        DatasetSubmission::STATUS_INCOMPLETE => 'Draft',
+        DatasetSubmission::STATUS_COMPLETE => 'Submitted',
+        DatasetSubmission::STATUS_IN_REVIEW => 'In Review',
+    );
+
+    /**
+     * The default action for Side by Side.
+     *
+     * @param Request     $request The Symfony request object.
+     * @param string|null $udi     The UDI of the Dataset to load.
+     *
+     * @Route("/{udi}")
+     *
+     * @return Response A Response instance.
+     */
+    public function defaultAction(Request $request, $udi = null)
+    {
+        if (!$this->isGranted('IS_AUTHENTICATED_FULLY')) {
+            return $this->redirect('/user/login?destination=' . $request->getPathInfo());
+        }
+        
+        if (!$this->isGranted(array('ROLE_DATA_REPOSITORY_MANAGER', 'ROLE_SUBJECT_MATTER_EXPERT'))) {
+            return $this->render('PelagosAppBundle:template:AdminOnly.html.twig');
+        }
+
+        return $this->render(
+            'PelagosAppBundle:SideBySide:index.html.twig'
+        );
+    }
+
+    /**
+     * The get versions action for Side by Side.
+     *
+     * @param Request $request The Symfony request object.
+     *
+     * @Route("/")
+     *
+     * @Method("POST")
+     *
+     * @return Response A Response instance.
+     */
+    public function getVersions(Request $request)
+    {
+        $udi = $request->request->get('udi');
+
+        try {
+            $datasetSubmissionHistory = $this->getDatasetSubmissionHistory($udi);
```

Comment:
Should we not get drafts? probably check for status incomplete?

Reply:
We want to get all of them. Including drafts.